### ISSUE TITLE
 Performance problems of eigh operator on CPU

### ISSUE THREAD

**Issue by Zjq9409** [pytorch/pytorch#65512]
## ❓ Questions and Help
`import numpy as np
from numpy import *
from torch.autograd import Variable
import time

random.seed(5)
N = 128
a = np.random.random(size=(N,N)).astype('float32')

times = 50

import torch
t_a = Variable(torch.from_numpy(a),requires_grad=True)
w, v = torch.linalg.eigh(t_a)
(w.sum() + abs(v).sum()).backward()
#print(w)
#print(t_a.grad)
start = time.time()
for i in range(times):
  w, v = torch.linalg.eigh(t_a)
  #(w.sum() + abs(v).sum()).backward()
print(">>>>>>>>>>>>>>pytorch>>>>>>>>>>>>>")
print( (time.time()-start)*1000/times ) #ms`

I use the following two commands on the CPU to run this code to test eigh performance:
(1) python test_cpu_eigh.py, this command will be used in multiple threads by default,The output time fluctuates obviously,Sometimes it’s 7.44ms, sometimes 12.88ms, or even 17.78ms.
(2)taskset -c 0 python test_cpu_eigh.py, this command is executed on a single core by default, at this time, the output time is relatively stable,always around 2.60ms.

I don't quite understand why this phenomenon occurs, why the single-core time is faster than the multi-core? Why is the multi-core execution time unstable?
